File: dataset_reduzido/codenet/Python/Code/s210889193.py

def solver():
	N = int(input())
	action = []
	for i in range(N):
		action.append(list(map(int, input().split())))

	dp = [[0 for _ in range(3)] for _ in range(N)]

	for i in range(3):
		dp[0][i] = action[0][i]

	for i in range(1,N):
		dp[i][0] = max(dp[i-1][1] + action[i][0], dp[i-1][2] + action[i][0])
		dp[i][1] = max(dp[i-1][0] + action[i][1], dp[i-1][2] + action[i][1])
		dp[i][2] = max(dp[i-1][0] + action[i][2], dp[i-1][1] + action[i][2])
	print(max(dp[N-1]))

solver()

File: output/StarCoder/codenet/Python/Java/s210889193.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Solver {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = Integer.parseInt(scanner.nextLine());
        List<List<Integer>> action = new ArrayList<>();
        
        for (int i = 0; i < N; i++) {
            String line = scanner.nextLine();
            List<Integer> row = new ArrayList<>();
            for (String s : line.split(" ")) {
                row.add(Integer.parseInt(s));
            }
            action.add(row);
        }
        
        Integer[][] dp = new Integer[N][3];
        
        for (int i = 0; i < 3; i++) {
            dp[0][i] = action.get(0).get(i);
        }
        
        for (int i = 1; i < N; i++) {
            dp[i][0] = Math.max(dp[i-1][1] + action.get(i).get(0), dp[i-1][2] + action.get(i).get(0));
            dp[i][1] = Math.max(dp[i-1][0] + action.get(i).get(1), dp[i-1][2] + action.get(i).get(1));
            dp[i][2] = Math.max(dp[i-1][0] + action.get(i).get(2), dp[i-1][1] + action.get(i).get(2));
        }
        
        System.out.println(Math.max(Math.max(dp[N-1][0], dp[N-1][1]), dp[N-1][2]));
        scanner.close();
    }
}